scipy/linalg/flapack_gen.pyf.src

```diff
@@ -221,6 +221,41 @@ subroutine <prefix>getrs(n,nrhs,lu,piv,b,info,trans)
 
 end subroutine <prefix>getrs
 
+subroutine <prefix>getc2(n,a,ipiv,jpiv,info)
+    ! lu,ipiv,jpiv,info = getc2(a,overwrite_a=0)
+    ! Compute an LU factorization of with complete pivoting of a general n-by-n matrix.
+    ! A = P * L * U * Q
+    threadsafe
+    callstatement {int i;(*f2py_func)(&n,a,&n,ipiv,jpiv,&info);for(i=0;i\<n;--ipiv[i],--jpiv[i++]);}
+    callprotoargument int*,<ctype>*,int*,int*,int*,int*
+
+    integer depend(a),intent(hide):: n = shape(a,0)
+    <ftype> dimension(n,n),intent(in,out,copy,out=lu) :: a
+    check(shape(a,0)==shape(a,1)) :: a
+    integer dimension(n),depend(n),intent(out) :: ipiv
```

Comment:
You need to add the lda here with the default value `MAX(1, shape(a, 0))` to avoid the segfaults with empty inputs. It is different than `n` hence you can't use `n` as a replacement. See LDA's explanation in the netlib docs.

Reply:
I was following the example of ?getrf. Is there a reason that ?getrf doesn't need `lda` but this does?

When I try using an empty array, I get:
```
ValueError: unexpected array size: new_size=1, got array with arr_size=0
```
which seems like a reasonable error message. It will segfault on some platforms?

I'll implement the suggestion; I'm just trying to understand.

Reply:
For example #7633